docs/src/main/sphinx/release/release-407.md

```diff
@@ -8,37 +8,96 @@
   `query.remote-task.request-size-headroom`, and
   `query.remote-task.guaranteed-splits-per-task` configuration properties or
   their respective session properties. ({issue}`15721`)
+* Add support for correlated queries with single-row values. ({issue}`15989`)
+* Improve query memory tracking. ({issue}`15983`)
 * Improve query performance when a predicate evaluates to a null value. ({issue}`15744`)
 * Improve performance of queries with joins on the output of global
   aggregations. ({issue}`15858`)
-* Fix precision loss when converting `time` values with a precision higher than
-  three and `time with time zone` values with lower precision. ({issue}`15861`)
+* Improve performance of selective queries, queries that read a small number of
+  columns, and queries that process tables with large Parquet row groups or Orc
+  stripes. ({issue}`15579`)
+* Improve concurrency when small, concurrent queries are run on a large cluster.
+  Trino now automatically determines many nodes to use for distributed joins
+  and aggregations. This is managed by the `query.max-hash-partition-count`
+  (renamed from `query.hash-partition-count`) and `query.min-hash-partition-count`
+  configuration properties. You can also now set the minimum number of rows and
+  minimum data size per task using the `optimizer.min-input-rows-per-task` and
+  `optimizer.min-input-size-per-task` configuration properties. ({issue}`15489`)
+* Improve performance of queries with window functions. ({issue}`15994`)
 * Change the `table_type` of materialized views in `information_schema.tables`
   from `BASE TABLE` to `MATERIALIZED VIEW`. ({issue}`15350`)
+* Fix precision loss when converting `time` values with a precision higher than
+  three and `time with time zone` values with lower precision. ({issue}`15861`)
+* Fix handling of `UNKNOWN` type for row type in exchange operator. ({issue}`15948`)
+* Fix potential incorrect results when a query fails but reports normal
```

Comment:
fails, but

Reply:
"reports normal completion" isn't an independent clause, so this comma would be incorrect